### PR TITLE
virtio_serial: fix 'NoneType' issue while transferring data

### PR DESCRIPTION
**Pull request by nanliu-r**
ID: 1776101

Signed-off-by: Nana Liu <nanliu@redhat.com>

### CODE REVIEW

**Comment by nanliu-r** [comment]
Test Pass.
 (09/10) repeat9.Host_RHEL.m8.u2.qcow2.virtio_blk.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.virtio_port_hotplug: PASS (268.95 s)
 (10/10) repeat10.Host_RHEL.m8.u2.qcow2.virtio_blk.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.virtio_port_hotplug: PASS (262.25 s)
RESULTS    : PASS 10 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0



@xiagao  @yanan-fu Please help to review this, thanks a lot.

File: qemu/tests/virtio_port_hotplug.py [hunk 1]
```diff
@@ -71,8 +74,16 @@ def run(test, params, env):
                 error_context.context("Load  module %s" % module, logging.info)
                 session.cmd("modprobe %s" % module)
         session.close()
+    check_pid_cmd = 'ps aux | grep "%s"| grep -v "grep"'
```
Comment (yanan-fu) [inline qemu/tests/virtio_port_hotplug.py]: `kill -9 `pgrep -f host_script``   or `pkill -f host_script`

Comment (nanliu-r) [reply qemu/tests/virtio_port_hotplug.py]: Thanks and update.

File: qemu/tests/virtio_port_hotplug.py [hunk 2]
```diff
@@ -71,8 +72,12 @@ def run(test, params, env):
                 error_context.context("Load  module %s" % module, logging.info)
                 session.cmd("modprobe %s" % module)
         session.close()
+    logging.info("Kill the first serial process on host")
+    host_script = params['host_script']
+    os.system('pkill -f %s' % host_script)
```
Comment (yanan-fu) [inline qemu/tests/virtio_port_hotplug.py]: Please use `process`, and check the status.

Comment (nanliu-r) [reply qemu/tests/virtio_port_hotplug.py]: Thanks and update.

File: qemu/tests/virtio_port_hotplug.py [hunk 3]
```diff
@@ -71,8 +72,14 @@ def run(test, params, env):
                 error_context.context("Load  module %s" % module, logging.info)
                 session.cmd("modprobe %s" % module)
         session.close()
+    logging.info("Kill the first serial process on host")
+    host_script = params['host_script']
+    result = process.system('pkill -f %s' % host_script, shell=True)
```
Comment (xiagao) [inline qemu/tests/virtio_port_hotplug.py]: If the original process is killed, the result is non 0.So, better to check if it existes.

Comment (nanliu-r) [reply qemu/tests/virtio_port_hotplug.py]: > If the original process finished, the result is non 0.So, better to check if it existes.

Thanks and update.

**Comment by yanan-fu** [comment]
LGTM
Please provide the latest test result, thanks!

**Comment by nanliu-r** [comment]
> LGTM
> Please provide the latest test result, thanks!


Test Pass:
 (09/10) repeat9.Host_RHEL.m8.u2.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.Win10.x86_64.io-github-autotest-qemu.virtio_port_hotplug: PASS (529.64 s)
 (10/10) repeat10.Host_RHEL.m8.u2.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.Win10.x86_64.io-github-autotest-qemu.virtio_port_hotplug: PASS (529.73 s)
RESULTS    : PASS 10 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0


 (1/4) Host_RHEL.m8.u2.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.Win2012.x86_64.io-github-autotest-qemu.unattended_install.cdrom.extra_cdrom_ks.default_install.aio_threads: PASS (2311.47 s)
 (2/4) Host_RHEL.m8.u2.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.Win2012.x86_64.r2.io-github-autotest-qemu.unattended_install.cdrom.extra_cdrom_ks.default_install.aio_threads: PASS (2988.02 s)
 (3/4) Host_RHEL.m8.u2.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.Win2012.x86_64.io-github-autotest-qemu.virtio_port_hotplug: PASS (438.16 s)
 (4/4) Host_RHEL.m8.u2.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.Win2012.x86_64.r2.io-github-autotest-qemu.virtio_port_hotplug: PASS (412.46 s)


(19/20) repeat10.Host_RHEL.m8.u2.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.Win2012.x86_64.io-github-autotest-qemu.virtio_port_hotplug: PASS (401.24 s)
 (20/20) repeat10.Host_RHEL.m8.u2.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.Win2012.x86_64.r2.io-github-autotest-qemu.virtio_port_hotplug: PASS (400.51 s)

**Pull request merged**